amxmodx/amx.cpp

```diff
@@ -4173,3 +4173,43 @@ int AMXAPI amx_GetStringOld(char *dest,const cell *source,int use_wchar)
   dest[len]='\0';     /* store terminator */
   return AMX_ERR_NONE;
 }
+
+
+#include <chrono>
+#include <amxmodx.h>
+#include <CPlugin.h>
+
+int AMXAPI amx_ExecPerf(AMX* amx, cell* retval, int index)
+{
+    CPluginMngr::CPlugin* perf_Plug = g_plugins.findPluginFast(amx);
+    if (amxmodx_perflog->value > 0.0f && perf_Plug && (perf_Plug->isDebug() || amxmodx_debug->value == 2.0f))
+    {
+        char perf_funcname[sNAMEMAX + 1];
+        perf_funcname[0] = '\0';
+        amx_GetPublic(perf_Plug->getAMX(), index, perf_funcname);
+        if (perf_funcname[0] == '\0')
+            sprintf(perf_funcname, "Unknown_ID%d", index);
+
+        const char* perf_plugname = perf_Plug->getName();
+        if (!perf_plugname || perf_plugname[0] == '\0')
+            perf_plugname = "Unknown_plugin";
+
+        using std::chrono::high_resolution_clock;
+        using std::chrono::duration_cast;
+        using std::chrono::duration;
+        using std::chrono::microseconds;
+
+        auto t1 = high_resolution_clock::now();
```

Comment:
You should use `steady_clock` instead. `high_resolution_clock` may be a subject to time adjustments.
See notes in https://en.cppreference.com/w/cpp/chrono/high_resolution_clock.

Reply:
Yes I see. Thanks

Reply:
@WPMGPRoSToTeMa  I can use "clock" function from https://www.cplusplus.com/reference/ctime/clock/ in this case ? Or this function has small resolution ?

Reply:
@WPMGPRoSToTeMa  

I use chrono because I think it has better precision, but if default function 'clock()' has normal precision, i can use it.

Reply:
I'm fine with using `chrono`, but let's wait for @dvander's opinion on this. As I remember `clock` has milliseconds precision on Windows, so it's not applicable here.

Reply:
@WPMGPRoSToTeMa okay, but at this time I update it to steady_clock https://github.com/alliedmodders/amxmodx/pull/1019/commits/58efc74c9e3e18ac123a24308394e5757bbfb73c

Reply:
Wait for dvander comment